Commit ID: 9f54dec7a11ce3bf32c27bb7b1f5ae015dd2f656
Change ID: lypmzsuqxonltkotsurqvyrvqlnlxorn
Author   : flukey <flukey@vapourmail.eu> (2026-10-05 15:33:29)
Committer: flukey <flukey@vapourmail.eu> (2026-10-05 15:58:35)

    mlockall default config

diff --git a/conf/example.conf.in b/conf/example.conf.in
index 57b3e5ffdf..76eda3c379 100644
--- a/conf/example.conf.in
+++ b/conf/example.conf.in
@@ -1883,7 +1883,7 @@
 	# Prior to version 2.02.62, LVM used mlockall() to pin the whole
 	# process's memory while activating devices.
 	# This configuration option has an automatic default value.
-	# use_mlockall = 0
+	# use_mlockall = 1
 
 	# Configuration option activation/monitoring.
 	# Monitor LVs that are activated.
diff --git a/lib/config/defaults.h b/lib/config/defaults.h
index d611453e65..86af960a0a 100644
--- a/lib/config/defaults.h
+++ b/lib/config/defaults.h
@@ -63,7 +63,11 @@
 #define DEFAULT_WAIT_FOR_LOCKS 1
 #define DEFAULT_LVMLOCKD_LOCK_RETRIES 6
 #define DEFAULT_PRIORITISE_WRITE_LOCKS 1
+#ifdef __GLIBC__
 #define DEFAULT_USE_MLOCKALL 0
+#else
+#define DEFAULT_USE_MLOCKALL 1
+#endif
 #define DEFAULT_METADATA_READ_ONLY 0
 #define DEFAULT_LVDISPLAY_SHOWS_FULL_DEVICE_PATH 0
 #define DEFAULT_UNKNOWN_DEVICE_NAME "[unknown]"
